c10/core/TensorImpl.cpp

```diff
@@ -115,10 +126,12 @@ TensorImpl* TensorImpl::maybe_zero_dim(bool condition_when_zero_dim) {
 }
 
 bool TensorImpl::has_storage() const {
-  return storage_;
+  TORCH_CHECK(type_id_ != UndefinedTensorId(), type_id_, " does not have has_storage");
+  return !is_sparse() && !is_mkldnn() && storage_;
```

Comment:
isn't this just strided?  Do we ever have strided tensors where "storage_" is false?